internal/output/combined.go

```diff
@@ -218,6 +220,16 @@ func Combine(inputs []ReportInput) (Root, error) {
 
 			diffTotalHourlyCost = decimalPtr(diffTotalHourlyCost.Add(*input.Root.DiffTotalHourlyCost))
 		}
+
+		if i != 0 && metadata.VCSRepoURL != input.Root.Metadata.VCSRepoURL {
+			logging.Logger.Warnf(
+				"combining Infracost JSON for different VCS repositories '%s' & '%s' this is likely to result in unexpected outputs",
```

Comment:
1. Should this warning just show once and list all the different repositories?
2. Should we state which repo we are using at the top level?

    e.g. "combining Infracost JSON for different VCS repositories 'infracost/repo1', 'infracost/repo2', infracost/repo3'. using 'infracost/repo1' as the top-level repository in the outputted JSON".

3. Should this warning only show if they are using JSON output?
4. Do we want to show a UI warning instead of log? I'm leaning towards no for just now since the logs will show for CI/CD which is probably where this matters.

Reply:
1& 2. Yep I think showing once with all the repos is good, I like messaging from output `2`. 
3. Probably yes.
4. I think let's probably stick it in UI. We want to highlight it to the user as much as possible.

Reply:
resolved in https://github.com/infracost/infracost/pull/1966/commits/72b1ca84e734ddcc7b28e9e8b03f35ec87231a47